### ISSUE TITLE
Provide "Getting Started" and Tutorials

### ISSUE THREAD

**Issue by i3games** [nannou-org/nannou#198]
Currently, "If you're new to nannou, we recommend checking out the examples to get an idea of how nannou applications are structured and how the API works."

This is a major roadblock for beginners (who may be new to Rust as well). 

It would be nice to have a "Getting started" (on the website), including Rust installation and setup and a number of tutorials by topic. 

Good example: https://processing.org/tutorials/

**Comment by madkeen** [comment]
Here are some good resources for getting started with Rust:

**Guides**
1. Small exercises to practice reading and writing Rust code! https://github.com/rust-lang/rustlings/
2. Rust by Example: https://doc.rust-lang.org/stable/rust-by-example/

**Books**
1. Official Rust Book: https://doc.rust-lang.org/book/second-edition/index.html
2. Porting C++ projects to Rust GitHub Book : https://locka99.gitbooks.io/a-guide-to-porting-c-to-rust/content/

**Online learning**
Udemy Rust Course (paid): https://www.udemy.com/rust-lang/)

**Comment by i3games** [comment]
Thanks for posting resources, very much appreciated. My intention with this issue was not so much for me learning the framework but for Nannou to become more widely known. 

A reason for the wide success of Processing in the creative coding community is IMHO due to the fact that from the start the team has balanced providing beginner-friendly tutorials, lots of code examples, a complete reference and a simple development environment and kept all that stuff updated and as complete as possible all the time.

**Comment by ELI7VH** [comment]
I would also appreciate a gentle handhold in regards to how to use more of nannou's framework.

Specifically, how to add targets, so we can do things like `cargo run --release --project my_project`
just the basics.

**Comment by amarao** [comment]
I'm a beginner to both nannou and Rust. I completely stuck between incomplete docs and complicated source code.

I want to display a graph of the equation I calculated in my buffer (simple vector with dimensions, as implemented by imgref). 

My problems:
1) I could not get my way through frame/window to use blit_from_simple_framebuffer (Should I use it? I just want to draw my 'picture' from buffer).
2) I can't find a good way to pass the buffer into 'view' function. The buffer takes up to few minutes to be calculated, so I don't want to do it in 'view' function. But nannou wants 'view' function, and it does not accept any closures (the error is expected type `for<'r> fn(&'r nannou::App, nannou::Frame) -> nannou::Frame`  found type `[closure@<::partial_application::partial macros>:5:1: 5:51 canvas:_]`).
3) There is no examples how to work with images at all.

**Comment by freesig** [comment]
Hey everyone.
We are currently in the process of pushing out the v0.9 release. It means some big changes to Nannou as we are switching to vulkan from OpenGL.
Everything is pretty much done and we are just waiting for some downstream crates to update. You can checkout the differences on the v0.9 branch. There is some image examples in the v0.9 branch however they are a bit low level.

Our next big goal is documentation and guides because it's definitely lacking currently.

> cargo run --release --project my_project

You want to do something like:
```
cargo new my_project
```
Then in the `Cargo.toml` add
```
[dependancies]
nannou = "0.8"
```

I would love to work with you all in the next few months to get feedback on guides, documentation and examples. I want to make it as easy as possible to learn Rust and Nannou.

__What you can do now__
Checkout the v0.9 branch
```
git fetch
git checkout v0.9
```
Build the new docs
```
cargo doc --open
```
Also have a look at the examples.
Run them and have a look over the code.
The vulkan examples are pretty low level and our plan is to wrap them in a higher level simpler API.

Jump on the nannou slack if you get stuck or create an issue. We might not be able to answer all questions immediately but it will help show use where documentation is most needed.

**Comment by tpltnt** [comment]
This can be closed given #309
